tests/src/e2e/ledger_tests.rs

```diff
@@ -401,98 +421,1522 @@ fn ledger_txs_and_queries() -> Result<()> {
+            ],
+            "Transaction is valid",
+        ),
+        // 4. Spend 7 BTC at SK(A) to PA(B)
+        (
+            vec![
+                "transfer",
+                "--source",
+                A_SPENDING_KEY,
+                "--target",
+                AB_PAYMENT_ADDRESS,
+                "--token",
+                BTC,
+                "--amount",
+                "7",
+                "--ledger-address",
+                &validator_one_rpc,
+            ],
+            "Transaction is valid",
+        ),
+        // 5. Spend 5 BTC at SK(B) to PA(C)
+        (
+            vec![
+                "transfer",
+                "--source",
+                B_SPENDING_KEY,
+                "--target",
+                AC_PAYMENT_ADDRESS,
+                "--token",
+                BTC,
+                "--amount",
+                "5",
+                "--ledger-address",
+                &validator_one_rpc,
+            ],
+            "Transaction is valid",
+        ),
+        // 6. Send 10 ETH from Bertha to PA(B)
+        (
+            vec![
+                "transfer",
+                "--source",
+                BERTHA,
+                "--target",
+                AB_PAYMENT_ADDRESS,
+                "--token",
+                ETH,
+                "--amount",
+                "10",
+                "--ledger-address",
+                &validator_one_rpc,
+            ],
+            "Transaction is valid",
+        ),
+    ];
 
-    client.assert_success();
-    let mut ledger = bg_ledger.foreground();
-    ledger.exp_string("some VPs rejected transaction")?;
+    // Wait till epoch boundary
+    let _ep0 = epoch_sleep(&test, &validator_one_rpc, 720)?;
 
-    // Wait to commit a block
-    ledger.exp_regex(r"Committed block hash.*, height: [0-9]+")?;
+    for (wallet_args, wallet_result) in wallet_args {
+            let mut client = run!(test, Bin::Wallet, wallet_args, Some(300))?;
+            client.exp_string(wallet_result)?;
+    }
 
-    // 3. Shut it down
-    ledger.send_control('c')?;
-    // Wait for the node to stop running to finish writing the state and tx
-    // queue
-    ledger.exp_string("Anoma ledger node has shut down.")?;
-    ledger.exp_eof()?;
-    drop(ledger);
+    for (tx_args, tx_result) in &txs_args {
+        let mut client = run!(test, Bin::Client, tx_args, Some(300))?;
+        if *tx_result == "Transaction is valid" {
+            if !cfg!(feature = "ABCI") {
+                client.exp_string("Transaction accepted")?;
+            }
+            client.exp_string("Transaction applied")?;
+        }
+        client.exp_string(tx_result)?;
+    }
 
-    // 4. Restart the ledger
+    // Get Albert's address from Wallet
+    let mut wallet_cmd = run!(test, Bin::Wallet, vec!["address", "find", "--alias", ALBERT], Some(300))?;
+    let wallet_output = wallet_cmd.exp_regex(r"Found address Established: .*")?;
+    let albert_address = &wallet_output.1[27..wallet_output.1.len()-1];
```

Comment:
there's a helper fn imported for this

```suggestion
    let albert_address = find_address(&test, ALBERT)?;
```